librz/il/definitions/bitvector.c

```diff
@@ -251,24 +251,48 @@ RZ_API RZ_OWN RzILBitVector *rz_il_bv_cut_tail(RZ_NONNULL RzILBitVector *bv, ut3
 }
 
 /**
- * Concat bv1 and bv2 to get new bitvector
+ * Concat bv2 to bv1 to get new bitvector
  * \param bv1 RzILBitVector
  * \param bv2 RzILBitVector
  * \return ret RzILBitVector, the new bitvector
  */
-RZ_API RZ_OWN RzILBitVector *rz_il_bv_concat(RZ_NONNULL RzILBitVector *bv1, RZ_NONNULL RzILBitVector *bv2) {
+RZ_API RZ_OWN RzILBitVector *rz_il_bv_append(RZ_NONNULL RzILBitVector *bv1, RZ_NONNULL RzILBitVector *bv2) {
 	rz_return_val_if_fail(bv1 && bv2 && bv1->bits && bv2->bits, NULL);
 
 	ut32 new_len = bv1->len + bv2->len;
 	RzILBitVector *ret = rz_il_bv_new(new_len);
 
 	// copy n bits from bv1
-	rz_il_bv_copy_nbits(bv1, 0, ret, 0, bv1->len);
-	rz_il_bv_copy_nbits(bv2, 0, ret, bv1->len, bv2->len);
+	rz_il_bv_copy_nbits(bv2, 0, ret, 0, bv2->len);
+	rz_il_bv_copy_nbits(bv1, 0, ret, bv2->len, bv1->len);
 
 	return ret;
 }
 
+/**
+ * Concat bitvectors from list bvs to get new bitvector
+ * \param bvs RzList of bitvectors
+ * \return ret RzILBitVector, the new bitvector
+ */
+RZ_API RZ_OWN RzILBitVector *rz_il_bv_concat(RZ_NONNULL RzList *bvs) {
+	rz_return_val_if_fail(bvs, NULL);
+	RzListIter *iter = bvs->head;
+	RzILBitVector *tmp = NULL, *ret = NULL, *bv = NULL;
+	rz_list_foreach(bvs, iter, bv){
+		if (iter == bvs->head){
+			ret = rz_il_bv_new(bv->len);
+			rz_il_bv_copy_nbits(bv, 0, ret, 0, bv->len);
+			tmp = ret;
+			continue;
+		}
+		ret = rz_il_bv_append(ret, bv);
+		rz_il_bv_free(tmp);
+		tmp = ret;
+	}
+	tmp = NULL;
+	return ret;
+}
```

Comment:
i'm not sure about this. @XVilka

Reply:
It actually works though. I can do it w/o append function, but it takes two foreach loops: 1st for collecting actual lengths, then we can allocate resulting bv, then we copy bvs content in the second foreach loop.

Reply:
looks correct, but the problem is if this is actually needed, or not. so this is why i asked to wait for him